plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -53,34 +59,86 @@ public void configure(Binder binder)
         configBinder(binder).bindConfig(OracleConfig.class);
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class)).setBinding().toInstance(ORACLE_MAX_LIST_EXPRESSIONS);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, RetryStrategy.class).setBinding().to(OracleRetryStrategy.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, Key.get(ConnectionFactory.class, ForRetryingConnectionFactory.class)).setBinding().toProvider(MyProvider.class).in(Scopes.SINGLETON);
```

Comment:
I was thinking that maybe that JDBC connector should provide: `@ForBaseJdbc ConnectionFactory.class` and `RetryStrategy.class` and `JdbcModule` would handle `ForRetryingConnectionFactory.class` binding. Like it handles `@StatsCollecting ConnectionFactory.class`, so we capture JMX stats for all JDBC connectors and specific JDBC connectors do not need to do anything special.

Reply:
I'm not sure I understand the `@StatsCollecting` use case properly, but you are saying

> so we capture JMX stats for **ALL** JDBC

The use case here is not for **ALL**, but for **SOME** we want to plug-in retries, but for **SOME** we want to leave it as it was. If I misunderstood, I will need more details on how exactly do you want me to simplify this.

Reply:
> The use case here is not for ALL

We discussed this offline and we decided it is for **ALL** connectors

Reply:
This is now implemented as agreed.